controllers/serving/inference_controller.go

```diff
@@ -305,8 +316,14 @@ func (ir *InferenceReconciler) syncServiceForInference(inf *servingv1alpha1.Infe
 		return ir.client.Create(context.Background(), &svc)
 	}
 
-	if !reflect.DeepEqual(svc.Spec, svcInCluster.Spec) {
-		svcInCluster.Spec = *svc.Spec.DeepCopy()
+	if !reflect.DeepEqual(svc.Spec, corev1.ServiceSpec{
```

Comment:
why is this update needed ?

and does this mean only these three fields will be compared ? Selector, Type, Ports

Reply:
yep,  I found that `svc.Spec` is always not deep-equal with `svcInCluster.Spec` because service will have a dynamically allocated ip in its spec after created.